plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -270,6 +273,14 @@ public String selectSql(TableId table, String formattedColumns)
         return format("SELECT %s FROM `%s`", formattedColumns, tableName);
     }
 
+    public void insert(InsertAllRequest insertAllRequest)
+    {
+        InsertAllResponse response = bigQuery.insertAll(insertAllRequest);
+        if (response.hasErrors()) {
+            throw new TrinoException(BIGQUERY_FAILED_TO_EXECUTE_QUERY, format("Failed to insert rows: %s", response.getInsertErrors()));
```

Comment:
Does BigQuery rollback already inserted rows or leaves table with partial inserts?

Reply:
It doesn't matter since we insert via temp tables but it's useful context to have to make sure future refactorings of direct insert mechanisms don't introduce bugs.

Reply:
It won't be rollbacked. Left a code comment.